smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -434,13 +434,57 @@ contract BridgeBank is BankStorage, CosmosBank, EthereumWhiteList, CosmosWhiteLi
       contractDecimals[token] = _decimals;
     } catch {
       // if we can't access the decimals function of this token,
-      // assume that it has 18 decimals
-      decimals = 18;
+      // assume that it has 0 decimals
+      decimals = 0;
```

Comment:
I think we also need to modify the section above (where we check decimals > 0)

Reply:
contractDecimals[token] is a mapping of token addresses to stored decimals from previous lookups. Since the default uint8 type will be 0, we must always check if it's something different than 0. If it's 0, it has two possible values, either 0 or not fetched, so we reach out to the contract and query the correct decimal number. If reaching out fails for any reason, we default to 0; otherwise, we return the result. I believe this is the correct way to handle it; let me know if not.